demo/api/src/importer/pipes/parsers/csv-parser.pipe.ts

```diff
@@ -54,3 +55,34 @@ export class CsvParsePipe implements ImporterPipe {
         };
     }
 }
+
+export class CsvParser extends Transform {
+    private parser: csv.CsvParserStream<csv.Row<unknown>, csv.Row<unknown>>;
+
+    constructor(private readonly options: CsvParserOptions) {
+        super({ objectMode: true });
+        this.parser = csv.parse({ ...this.options });
+    }
+
+    _transform(input: { chunk: Buffer | string; metadata: Record<string, unknown> }, encoding: BufferEncoding, callback: TransformCallback) {
+        const { chunk, metadata } = input;
+
+        this.parser.removeAllListeners();
```

Comment:
Why is it needed?  What happens with the attached listeners coming from `this.options`?

Reply:
Are they removed and attached (again) after each chunk of data?  This could cause performance issues, I think.